src/controllers/contract/index.ts

```diff
@@ -199,26 +199,29 @@ export const contractDeployCommand: Command = {
     }) {
         let account = await getAccount(terminal, args);
         const info = await account.getAccount();
+        const accountAddress = await account.getAddress()
         if (info.acc_type === AccountType.active) {
-            throw new Error(`Account ${await account.getAddress()} already deployed.`);
+            throw new Error(`Account ${accountAddress} already deployed.`);
         }
-        const network = new NetworkRegistry().get(args.network);
+        const { giver: giverInfo, name: networkName } = new NetworkRegistry().get(args.network);
         const currentBalance = BigInt(info.balance ?? 0);
-        const requiredBalance = parseNanoTokens(args.value) ?? network.giver?.value ?? DEFAULT_TOPUP_VALUE;
-
-        if (currentBalance < BigInt(requiredBalance)) {
-            const giverInfo = new NetworkRegistry().get(args.network).giver;
-            if (!giverInfo) {
-                throw new Error(`Account ${await account.getAddress()} has low balance to deploy.\n` +
-                    `You have to create an enough balance before deploying in two ways: \n` +
-                    `sending some value to this address\n` +
-                    `or setting up a giver for the network with \`tondev network giver\` command.`,
-                );
-            }
-            const giver = await NetworkGiver.get(account.client, giverInfo);
-            giver.value = requiredBalance;
-            await giver.sendTo(await account.getAddress(), requiredBalance);
-            await giver.account.free();
+        const explicitTopUp = parseNanoTokens(args.value)
+        const implicitTopUp = currentBalance === BigInt(0) ? giverInfo?.value || DEFAULT_TOPUP_VALUE : 0
+
+        if (explicitTopUp && !giverInfo) {
+            throw new Error(
+                `A top-up was requested, but no giver for the network ${networkName} was found.\n` +
+                    `You have to set up a giver for this network with \`tondev network giver\` command.`,
+            )
+        }
+
+        const topUp = explicitTopUp || implicitTopUp
+
+        if (giverInfo && topUp) {
+            const giver = await NetworkGiver.get(account.client, giverInfo)
```

Comment:
I think we must rename `NetworkGiver.get` to `NetworkGiver.create` because `get` means that some existing instance can be returned.

Reply:
Done.  Method renamed